### PR TITLE
Get function category names

### PR DESCRIPTION
**Pull request by anderson-joyle**
Addressing part 1 of https://github.com/microsoft/Power-Fx/issues/2151.
> ...extend the function info, with the localized function category names.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/FunctionTests.cs [hunk 1]
```diff
@@ -36,6 +38,42 @@ public void TestAllBuiltinFunctionsHaveParameterDescriptions()
...
+                    {
+                        Assert.Contains(StringResources.Get(category.ToString(), locale_enus.Name), categoriesNames_enus);
+                        Assert.Contains(StringResources.Get(category.ToString(), locale_ptbr.Name), categoriesNames_ptbr);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void FunctionCategoriesResourcesTest()
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/FunctionTests.cs:67]: >FunctionCategoriesResourcesTest [](http://example.com/codeflow?start=20&length=31)

this is good.

File: src/tests/Microsoft.PowerFx.Core.Tests/FunctionTests.cs [hunk 2]
```diff
@@ -36,6 +38,42 @@ public void TestAllBuiltinFunctionsHaveParameterDescriptions()
...
+        [Fact]
+        public void FunctionCategoriesLocaleTest()
+        {
+            var texlFunctionsLibrary = BuiltinFunctionsCore.BuiltinFunctionsLibrary;
+            var locale_enus = CultureInfo.CreateSpecificCulture("en-US");
+            var locale_ptbr = CultureInfo.CreateSpecificCulture("pt-BR");
+
+            foreach (var func in texlFunctionsLibrary)
+            {
+                var info = new FunctionInfo(func);
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/FunctionTests.cs:50]: I don't think we need to call on every function in the library - lots of overlap. 
We can just call on one specific one - and assert the actual contents. 

Can we move this to FunctionInfoTests.cs ?

File: src/strings/PowerFxResources.en-US.resx [hunk 3]
```diff
@@ -2398,6 +2398,9 @@
     <value>Invalid json pointer.</value>
     <comment>Generic json pointer parsing error.</comment>
   </data>
+  <data name="None" xml:space="preserve">
+    <value>None</value>
```
Comment (CarlosFigueiraMSFT) [inline src/strings/PowerFxResources.en-US.resx:2402]: >None [](http://example.com/codeflow?start=11&length=4)

Please add comments explaining the context in which this value will be displayed to the user (and preferably to the other types as well). Without them the translators don't have any context on what the term means, which will cause bad translations (e.g., wrong gender, wrong syntactical category, ...)

Comment (CarlosFigueiraMSFT) [reply src/strings/PowerFxResources.en-US.resx:2402]: For example, this is being used as a function category, which should be a feminine noun in Portuguese, but the current translation for the term Logical (2 down) is a masculine adjective.

**Comment by CarlosFigueiraMSFT** [comment]
>     <value>Date &amp; Time</value>

I know this was already there, but can you change it to `Date and time` or `Date & time`? We use sentence casing throughout the platform. Same with `Math & statistical` below

---
Refers to: src/strings/PowerFxResources.en-US.resx:2417 in 55e6a55. [](commit_id = 55e6a55d159464430c547477a354b99e74a3dd6f, deletion_comment = False)

File: src/libraries/Microsoft.PowerFx.Core/Functions/Utils.cs [hunk 4]
```diff
@@ -15,5 +18,10 @@ public static bool TestBit(this BigInteger value, int bitIndex)
 
             return !(value & (BigInteger.One << bitIndex)).IsZero;
         }
+
+        public static string GetLocalizedName(this FunctionCategories category, CultureInfo culture)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Functions/Utils.cs:22]: >GetLocalizedName [](http://example.com/codeflow?start=29&length=16)

Nice! Extension method is sweet

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/strings/PowerFxResources.en-US.resx [hunk 5]
```diff
@@ -2398,37 +2398,54 @@
...
+    <comment>Function category name</comment>
```
Comment (CarlosFigueiraMSFT) [inline src/strings/PowerFxResources.en-US.resx:2403]: >Function category name [](http://example.com/codeflow?start=13&length=22)

Suggestion for names and descriptions

- None: Function category name - function without pre-defined categories
- Text: Function category name - function that works on text values
- Logical: Function category name - function that works on Boolean values
- Table: Function category name - function that works on tabular data
- Behavior: Function category name - function that needs to be executed in behavior contexts (i.e., not purely functional, with side effects)
- Date and time: Function category name - function that works on date/time values
    - Please use "Date and ***time*** (sentence casing)
- Math and statistical: Function category name - functions for mathematical and/or statistical operations
    - Please use "Math and ***statistical*** (sentence casing)
- User-defined: Function category name - function that is defined by the user, not a built-in function
- Information: Function category name - function that returns information about the environment where it is running
- Color: Function category name - function that works on or returns color values
- Service: Function category name - function that comes from a web service, typically using REST (REpresentational State Transfer) communication
- Component: Function category name - function defined in components within the environment where it is running

**Review by MikeStall — DISMISSED** [review]
:shipit:

File: src/strings/PowerFxResources.en-US.resx (hunk 5 above)
Comment (anderson-joyle) [reply src/strings/PowerFxResources.en-US.resx]: Great suggestions. Thanks.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
🚢

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**